### PR TITLE
Add support for senseval3

### PR DESCRIPTION
**Pull request by Ayushk4**
- [X] Support for Senseval3 corpora
- [X] Documentation
- [x] Tests
 Refer #9 

Some differences from SemCor -
- The structure is `document, sentences, words` rather than `document, paragraph, sentences, words`. The changes were made accordingly.
- The Regex expression was relaxed to accommodate certain cases of `lexsn='U','NOANSWER','NOWN'`

### CODE REVIEW

File: src/Senseval3.jl [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        "<wf cmd=ignore"=>       get_tagged,
+        "<wf cmd=done"=>         line -> contains(line,"lemma=") ? get_sense_annotated(line) : get_tagged(line),
+        "<punc>"=>               get_punc,
+        "<context"=>             ignore,
+        "</context" =>           ignore,
+        "<s" =>                  new_sentence,
+        "</s" =>                 ignore
+    ]
+
+
```
Comment (oxinabox) [inline src/Senseval3.jl]: Lets move the bit below (and its match in the SemCode file into

```
"""
    apply_subparsers(filename, subparsers)

Applies subparses to each line of the file.
`subparsers` should be a `Vector{Pair{String,Function}}`,
or similar ordered mapping from a line prefix, to a closure for how to parse that line.
This function does not return anything, instead the subparser closures (functions),
should mutate the state in the environment they are defined.
"""
function apply_subparsers(filename, subparsers)
 for line in eachline(filename)
        try
            found = false
            for (prefix, subparse) in subparsers
                if startswith(line, prefix)
                    found=true
                    subparse(line)
                    break
                end
            end
            @assert(found, "No parser for \"$line\"")
        catch ee
            warn("Error parsing \"$line\". $ee")
            rethrow()
        end
        return nothing
    end
```
```

**Comment by Ayushk4** [comment]
Do we need a separate testset for apply_parsers function? If yes, then we would have to export it for testing.

**Comment by oxinabox** [comment]
I don't think we really need a seperate test for it.
This package doesn't have extreme standards for small unit tests.

(Also for testing nonexpected functions you just do `using CorpusLoaders: apply_parsers`, or `CorpusLoaders.apply_parsers` in the tests)

File: docs/src/SemCor.md [hunk 2]
```diff
@@ -15,70 +15,65 @@ In the demo below is is shown how one might index the corpus to get a collection
 ```julia
```
Comment (oxinabox) [inline docs/src/SemCor.md]: good catch, this became outdated, with the InternedStrings changes

Comment (Ayushk4) [reply docs/src/SemCor.md]: I feel now, that I should have commented that fact while updating. :sweat_smile: 
Its my bad.

**Comment by Ayushk4** [comment]
@oxinabox I think that I am done with this from my side. You may review the PR.

File: src/Senseval3.jl [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        paths = joinpath.(dirpath, readdir(dirpath))
+    end
+    Senseval3(paths)
+end
+
+Senseval3() = Senseval3(datadep"Senseval 3")
+
+MultiResolutionIterators.levelname_map(::Type{Senseval3}) = [
+    :doc=>1, :contextfile=>1, :context=>1,
+    :sent=>2,
```
Comment (oxinabox) [inline src/Senseval3.jl]:
```suggestion
    :sent=>2, :sentence=>2,
```

File: src/Senseval3.jl [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    :word=>3, :token=>4,
```
Comment (oxinabox) [inline src/Senseval3.jl]:
```suggestion
    :word=>3, :token=>3,
```

Is this mistake in SemCore also? Words and tokens are the same thing

File: src/Senseval3.jl [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    :char=>5]
```
Comment (oxinabox) [inline src/Senseval3.jl]:
```suggestion
    :char=>4]
```

File: src/Senseval3.jl (hunk 4 above)
Comment (Ayushk4) [reply src/Senseval3.jl]: Yes. I got confused with that. They should be the same thing. I will make the changes on both.

File: src/Senseval3.jl [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    SenseAnnotatedWord(pos, lemma, parse(Int, wnsn), lexsn, word)
+end
+
+
+function parse_senseval3file(filename)
+    local sent
+    lines = @NestedVector(TaggedWord,2)()
+    context = Document(intern(basename(filename)), lines)
+
+    ignore(line) = nothing
```
Comment (oxinabox) [inline src/Senseval3.jl]: move `ignore` to the `apply_subparsers` file,
and document it as a helper for ignoring lines when using subparsers.
Also delete it from semcor

**Comment by oxinabox** [comment]
Looks pretty good.
A few small things, 
that might actually be mistakes in the SemCore implementation too.

**Comment by oxinabox** [comment]
Awesome, onces tests pass, can merge

**Comment by oxinabox** [comment]
LGTM, thanks

**Comment by Ayushk4** [comment]
Thank you for your guidance on this. I learned a lot while working on this. :slightly_smiling_face:

**Pull request merged**